Hair Chems

Vimas21 opened the pull request:
:cl: Vimas
rcsadd: Capillium, a reagent that grows hair on touch.
rscadd: Hair dye, a reagent that, when mixed with crayon dust, can dye hair on touch.
/:cl:

SammasC123 committed:
Hair Chems

We have two big additions (Not really that big)

Hair growth serum- capillium
And hair dye!

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -1018,3 +1018,4 @@
 	..()
 	M.add_chemical_effect(CE_TOXIN, 1)
 	M.immunity -= 0.5 //inverse effects when abused
+
```
Line break change but no other changes in this file - Intentional?

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm:
```diff
@@ -469,6 +469,73 @@
+
+/datum/reagent/capilliumate/affect_touch(var/mob/living/carbon/human/M, var/alien, var/removed)
+	if(!alien)
+		M.change_hair("Very Long Hair")
```
This should probably have some kind of message displayed indicating what happened.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm:
```diff
@@ -469,6 +469,73 @@
+	heating_products = null
+	heating_point = null
+
+/datum/reagent/colored_hair_dye
```
Having two different hair_dye paths seems redundant - You can probably safely merge these into a single type path

Vimas21 replied:
So, the base reaction to color the dye is [hair dye + color] = colored hair dye

If it became hair_dye/color, the reaction would trigger again, wouldn't it? Going through cyclically.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm:
```diff
@@ -469,6 +469,73 @@
+		var/red = hex2num(copytext(src.color, 2, 4))
+		var/green = hex2num(copytext(src.color, 4, 6))
+		var/blue = hex2num(copytext(src.color, 6, 8))
+		M.change_hair_color(red, green, blue)
```
Same as before - Could probably use a message to indicate what happened.

Spookerton commented on code/modules/reagents/Chemistry-Recipes.dm:
```diff
@@ -632,6 +632,13 @@
+/datum/chemical_reaction/capilliumate
+	name = "Capilliumate"
+	result =/datum/reagent/capilliumate
+	required_reagents = list(/datum/reagent/radium = 1, /datum/reagent/blood = 1, /datum/reagent/mutagen = 1)
```
go with `/datum/reagent/nutriment/protein` instead of `/datum/reagent/blood`.

SammasC123 committed:
Bugfixes

SammasC123 committed:
Protein Fix

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm:
```diff
@@ -469,6 +469,75 @@
+	if(!alien)
+		M.change_hair("Very Long Hair")
+		M.change_facial_hair("Very Long Beard")
+		to_chat(M, "Your hair grows to extraordinary lengths!")
```
I would also recommend using a span in these messages to help differentiate it from all the other black text from chat - `SPAN_NOTICE()` seems appropriate for this type of message.

`to_chat(M, SPAN_NOTICE("Your hair grows to extraordinary lengths!"))`

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm:
```diff
@@ -469,6 +469,75 @@
+		var/green = hex2num(copytext(src.color, 4, 6))
+		var/blue = hex2num(copytext(src.color, 6, 8))
+		M.change_hair_color(red, green, blue)
+		to_chat(M, "Your hair suddenly changes color!")
```
I would also recommend using a span in these messages to help differentiate it from all the other black text from chat - SPAN_NOTICE() seems appropriate for this type of message.

See example provided above.

SierraKomodo reviewed (changes requested):
Alongside the other items I would like to start seeing codex entries added for things that are being added/modified - If you need help with setting up codex stuff, give me a poke in `#coding` on discord

This will also need squashing before it can be approved for merging - https://github.com/Baystation12/Baystation12/wiki/Squashing

SierraKomodo commented:
Disregard the codex notes - It won't be possible to define these explicitly due to the current state of category_reagents.dm until #29167 is merged.

SammasC123 committed:
Hair Chems

We have two big additions (Not really that big)

Hair growth serum- capillium
And hair dye!
Bugfixes


Protein Fix


Span_notice

To make it more noticable
Also it affects beards now

Good catch Sierra

SammasC123 committed:
Merge remote-tracking branch 'Vimas21/ChemicalAdditions' into ChemicalAdditions

I'm screaming
